### DISCUSSION TITLE
Deserializing an Option as None when field is missing

### DISCUSSION THREAD

**Discussion by petejodo** [arrow-kt/arrow#3115] [Q&A]
I'm not sure this is actually possible so asking as a question here but given the following:

```kotlin
@Serializable
data class PatchData(
  val required: Int,
  val someNullableProperty: Option<Int?>,
)
```

Is it possible to have `someNullableProperty` to only be `None` if the field is missing from the json but `Some(null)` if the field is set to `null`? E.g.

- `{ "required": 1 }` becomes `PatchData(required: 1, someNullableProperty: None)`
- `{ "required": 1, "someNullableProperty": null }` becomes `PatchData(required: 1, someNullableProperty: Some(null))`

This is to deal with PATCH requests where the field missing represents something different than when the field is null.

I know this is probably counter to the ideal of arrow but figured I'd ask. Also would be interesting to hear how others may represent/handle this. Thanks

**Comment by kyay10** [comment]
It suffices to use a default value:
```kotlin
@Serializable
data class PatchData(
  val required: Int,
  val someNullableProperty: Option<Int?> = None,
)
```
